### PR TITLE
AKU-370: Manage Aspects action updates

### PR DESCRIPTION
**Pull request by draperd**
This PR addresses https://issues.alfresco.com/jira/browse/AKU-370 and https://github.com/Alfresco/Aikau/issues/299 as well as making general improvements to Aikau action handling for actions derived from node metadata.

### CODE REVIEW

File: aikau/src/main/resources/alfresco/renderers/actions/ManageAspects.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * <p>This module provides the configuration that can be used for rendering a [menu item]{@link module:alfresco/menus/AlfMenuItem}
+ * for the "Manage Aspects" action. The main thing this provides is the "renderFilters" configuration that can be
+ * used to determine whether or not the action should be displayed based on the attributes of the "currentItem" (e.g. the Node)
+ * that the action is to be used against.</p>
+ * <p>It should be possible to upload a new version of a Node when the following is true:
```
Comment (MartinDoyleUK) [inline aikau/src/main/resources/alfresco/renderers/actions/ManageAspects.js]: Missing closing `</p>`

File: aikau/src/main/resources/alfresco/services/actions/ManageAspectsService.js [hunk 2]
```diff
@@ -129,32 +129,27 @@ define(["dojo/_base/declare",
...
-               array.forEach(payload.item.node.aspects, lang.hitch(this, this.processAspect, aspects));
-               this.showAspectsDialog(payload.item, aspects);
-            }
-            else
-            {
-               // We need to request the aspects for the current node...
-               this.onAspects(payload.item);
-            }
+            this.serviceXhr({
+               url: AlfConstants.PROXY_URI + "slingshot/doclib/aspects/node/" + payload.node.nodeRef.replace("://", "/"),
```
Comment (MartinDoyleUK) [inline aikau/src/main/resources/alfresco/services/actions/ManageAspectsService.js]: Are we certain `payload.node.nodeRef` is a `string` at this point?

File: aikau/src/main/resources/alfresco/services/actions/ManageAspectsService.js [hunk 3]
```diff
@@ -329,7 +307,8 @@ define(["dojo/_base/declare",
             added: added,
             removed: removed
          };
-         this.serviceXhr({url: AlfConstants.PROXY_URI + "slingshot/doclib/aspects/node/" + payload.item.nodeRef,
+         var processedNodeRef = payload.item.node.nodeRef.replace("://", "/");
```
Comment (MartinDoyleUK) [inline aikau/src/main/resources/alfresco/services/actions/ManageAspectsService.js]: Again, are we certain about the `nodeRef` being a `string`?

Comment (draperd) [inline aikau/src/main/resources/alfresco/services/actions/ManageAspectsService.js]: Yep, 100%

File: aikau/src/main/resources/alfresco/services/actions/ManageAspectsService.js (hunk 2 above)
Comment (draperd) [inline aikau/src/main/resources/alfresco/services/actions/ManageAspectsService.js]: Yes, very much so... if it's not, we're in serious trouble long before this point :)

**Pull request merged**